smartsim/_core/control/controller.py

```diff
@@ -408,7 +418,7 @@ def _launch_step(self, job_step, entity):
             logger.debug(f"Launching {entity.name}")
             self._jobs.add_job(job_step.name, job_id, entity, is_task)
 
-    def _create_batch_job_step(self, entity_list):
+    def _create_batch_job_step(self, entity_list: EntityList) -> t.Any:
```

Comment:
That call-tree is out of scope for the public API. I use the minimum to avoid mypy complaining about non-typed methods called in a typed context for now. 

It will be fixed in my next PR when I add the typehints for the remaining code.